### PR TITLE
[forge] Add MempoolToBlockCreation, rm QsBatchToPos, QsPosToProposal

### PR DESCRIPTION
**Pull request by hariria**
## Description

Forge provides too much granularity with 4 columns capturing metrics for all parts the blockchain latency in 4 components. The goal of this PR is to simplify this into 3 columns.

Ultimately the table will have the following LatencyBreakdownSlice metrics (ignoring indexer metrics):

- `InsertionToBlock` -> renamed to `MempoolToBlockCreation` (`mempool->block`)
    - merges `QsBatchToPos` and `QsPosToProposal`
- `ConsensusProposalToOrdered`  (no change)
- `ConsensusOrderedToCommit`  (no change) 

### OLD
batch-pos

### New

Screenshot 2024-10-21 at 2 48 27 PM

## How Has This Been Tested?

Tested queries on grafana 👍 

- [Adhoc Forge Run before](https://github.com/aptos-labs/aptos-core/actions/runs/11264334213)
- [SUCCESS][`realistic_env_load_sweep`][Adhoc Forge Run after (with `realistic_env_load_sweep`)](https://github.com/aptos-labs/aptos-core/actions/runs/11450618910)
- [SUCCESS][`realistic_env_workload_sweep`][Adhoc Forge Run after (with `realistic_env_workload_sweep`)](https://github.com/aptos-labs/aptos-core/actions/runs/11467951186)

## Evidence for latency limit changes (non-exhaustive)
- [FAILURE][`realistic_env_load_sweep`] [`Failed latency check, for \["P90 latency for 10 is 0.9s and exceeds limit of 0.9s"\]`](https://github.com/aptos-labs/aptos-core/actions/runs/11449840312)
- [FAILURE][`realistic_env_workload_sweep`] [`"ConsensusOrderedToCommit for 0: CoinTransfer\[B:20.0k\]" metric violated threshold of 0.4, max_breach_pct: 0, breach_pct: 100 `](https://github.com/aptos-labs/aptos-core/actions/runs/11451182248)
- [FAILURE][`realistic_env_workload_sweep`] [`"ConsensusOrderedToCommit for 3: ph1: TokenV2AmbassadorMint\[B:20.0k\]" metric violated threshold of 0.6, max_breach_pct: 0, breach_pct: 50 `](https://github.com/aptos-labs/aptos-core/actions/runs/11464240477)
- [FAILURE][`realistic_env_workload_sweep`][`"ConsensusOrderedToCommit for 3: ph1: TokenV2AmbassadorMint\[B:20.0k\]" metric violated threshold of 0.7, max_breach_pct: 0, breach_pct: 33 `](https://github.com/aptos-labs/aptos-core/actions/runs/11464967638)

### CODE REVIEW

File: testsuite/forge/src/interface/prometheus_metrics.rs [hunk 1]
```diff
@@ -158,6 +159,12 @@ pub async fn fetch_latency_breakdown(
     let qs_batch_to_pos_query = r#"sum(rate(quorum_store_batch_to_PoS_duration_sum{role=~"validator"}[1m])) / sum(rate(quorum_store_batch_to_PoS_duration_count{role=~"validator"}[1m]))"#;
     let qs_pos_to_proposal_query = r#"sum(rate(quorum_store_pos_to_pull_sum{role=~"validator"}[1m])) / sum(rate(quorum_store_pos_to_pull_count{role=~"validator"}[1m]))"#;
 
+    // the result of mempool_to_block_creation_query is the sum of the individual limit values of qs_batch_to_pos_query and qs_pos_to_proposal_query
+    let mempool_to_block_creation_query = format!(
+        "sum({}, {})",
+        qs_batch_to_pos_query, qs_pos_to_proposal_query
+    );
```
Comment (igor-aptos) [inline testsuite/forge/src/interface/prometheus_metrics.rs:166]: I think @zekun000 was mentioning that this sum was not precise, and that the query you used in the previous version of the PR is computing things better?

File: testsuite/forge/src/interface/prometheus_metrics.rs [hunk 2]
```diff
@@ -108,6 +108,7 @@ pub async fn fetch_system_metrics(
 pub enum LatencyBreakdownSlice {
     QsBatchToPos,
     QsPosToProposal,
```
Comment (igor-aptos) [inline testsuite/forge/src/interface/prometheus_metrics.rs:110]: remove these two, and then you'll see what else you'll need to modify

this enum is not serialized, so we are free to remove elements from here

**Review by igor-aptos — CHANGES_REQUESTED** [review]
also remove the lines you commented out

File: testsuite/forge/src/interface/prometheus_metrics.rs (hunk 1 above)
Comment (hariria) [reply testsuite/forge/src/interface/prometheus_metrics.rs:164]: > "...update all tests that use QsBatchToPos / QsPosToProposal, to use new MempoolToBlockCreation instead (just sum up their individual limit values together"

looking at our chats... I was under the impression you wanted me to do calculate the sum, but i can change to the previous query

File: testsuite/forge/src/interface/prometheus_metrics.rs (hunk 2 above)
Comment (hariria) [reply testsuite/forge/src/interface/prometheus_metrics.rs]: Sounds good 👍

File: testsuite/forge-cli/src/suites/realistic_environment.rs [hunk 3]
```diff
@@ -323,9 +324,7 @@ pub(crate) fn realistic_env_max_load_test(
         success_criteria = success_criteria.add_latency_breakdown_threshold(
             LatencyBreakdownThreshold::new_with_breach_pct(
                 vec![
-                    (LatencyBreakdownSlice::QsBatchToPos, 0.35),
-                    // quorum store backpressure is relaxed, so queueing happens here
-                    (LatencyBreakdownSlice::QsPosToProposal, 2.5),
+                    (LatencyBreakdownSlice::MempoolToBlockCreation, 0.35 + 2.5),
```
Comment (igor-aptos) [inline testsuite/forge-cli/src/suites/realistic_environment.rs:327]: add back the comment for
// quorum store backpressure is relaxed, so queueing happens here

to explain the high limit

**Review by igor-aptos — APPROVED** [review]
thanks!

File: testsuite/forge-cli/src/suites/realistic_environment.rs (hunk 3 above)
Comment (hariria) [reply testsuite/forge-cli/src/suites/realistic_environment.rs:328]: Done!

**Review by zekun000 — APPROVED** [review]

**Pull request merged**